src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -909,7 +909,7 @@ private static int getGroupNumber(Pattern[] patterns, String name) {
                     bestEnd = matcher.end();
                     bestPos = matcher.start();
                 }
-                else if (matcher.start() == bestPos && matcher.end() > bestEnd) {
+                else if (bestEnd < matcher.end()) {
```

Comment:
Please make it `matcher.end() > bestEnd`. Looks more readable this way

Reply:
That's what I used for diff report:

```groovy diff.groovy --localGitRepo ../../../IdeaProjects/checkstyle --baseBranch master --patchBranch issue_7872_hardcode --config my_check.xml --listOfProjects projects-to-test-on.properties```

`cat my_check.xml`
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
    <property name="severity" value="warning"/>

    <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
    <property name="haltOnException" value="false"/>

    <!-- BeforeExecutionFileFilters is required for sources of java9 -->
    <module name="BeforeExecutionExclusionFileFilter">
        <property name="fileNamePattern" value="module\-info\.java$" />
    </module>

    <module name="TreeWalker">
	<module name="ImportOrder">
  	    <property name="groups" value="*,javax,java"/>
  	    <property name="ordered" value="true"/>
  	    <property name="separated" value="false"/>
  	    <property name="option" value="bottom"/>
  	    <property name="sortStaticImportsAlphabetically" value="true"/>
	</module>
	<module name="SuppressionXpathSingleFilter">
  	    <property name="checks" value="ImportOrder"/>
      	    <property name="message" value="^'java\..*'.*"/>
	</module>
    </module>
    <!-- Example of filter -->
    <!--
    <module name="SeverityMatchFilter">
        <property name="severity" value="warning"/>
        <property name="acceptOnMatch" value="false"/>
    </module>
    -->

    <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
    <module name="SuppressionSingleFilter">
      <property name="message" value="Exception occurred while parsing"/>
      <property name="checks" value="Checker"/>
    </module>
</module>

```

Reply:
Do I need to change from:
`<property name="groups" value="*,javax,java"/>`
to
`<property name="groups" value="*,javax, java, java.*"/>` or for something else? 

I don't know any other way of killing mutation without removing this check `matcher.start() == bestPos`. Do you know any other possible solutions to kill mutation?